mobile/test/common/integration/test_server_interface.cc

```diff
@@ -9,14 +9,13 @@ static std::weak_ptr<Envoy::TestServer> weak_test_server_;
 
 static std::shared_ptr<Envoy::TestServer> test_server() { return weak_test_server_.lock(); }
 
-void start_server(bool use_quic) {
+void start_server(bool use_quic, bool disable_https) {
   Envoy::ExtensionRegistry::registerFactories();
   strong_test_server_ = std::make_shared<Envoy::TestServer>();
   weak_test_server_ = strong_test_server_;
 
-  if (auto e = test_server()) {
-    e->startTestServer(use_quic);
-  }
+  auto e = test_server();
```

Comment:
I think, though not 100% sure, the reason we use a weak_ptr here is because if someone calls `shutdown_server()` before `start_server()` has finished running, then the reference pointed to by the weak_ptr will no longer be valid.  In which case, we should keep the nullptr check here and below.

If I'm wrong and we don't need to check for nullptr, then this code could just be inlined, i.e. `test_server()->startTestServer(...)`.

Reply:
> But it also seems fine to fail in this case? 

I think it depends on what the desired behavior is.  The instances of using the test server that I could find (all in Java land) call everything in the expected order.  But if that's the expected behavior, I'm not sure I understand what the weak_ptr is doing.

Reply:
I think the idea is maybe to reuse the test server across threads / tests without having to pass it around?

Reply:
if that's the case, I think we still want the nullptr check.  Probably best to keep the null check as it was, and rename the variable as Ryan suggested to something more meaningful